python/meep.i

```diff
@@ -418,11 +418,14 @@ meep::volume_list *make_volume_list(const meep::volume &v, int c,
 template<typename dft_type>
 PyObject *_get_dft_array(meep::fields *f, dft_type dft, meep::component c, int num_freq) {
     int rank;
-    size_t dims[3];
+    size_t dims[3] = {1,1,1};
     std::complex<double> *dft_arr = f->get_dft_array(dft, c, num_freq, &rank, dims);
 
-    if (rank==0 || dft_arr==NULL) // this can happen e.g. if component c vanishes by symmetry
-     return PyArray_SimpleNew(0, 0, NPY_CDOUBLE);
+    if (rank==0 || dft_arr==NULL) { // this can happen e.g. if component c vanishes by symmetry
+        npy_intp di[1] = {1};
+        std::complex<double> d[1] = {std::complex<double>(0,0)};
+        return PyArray_SimpleNewFromData(1, di, NPY_CDOUBLE, d);
```

Comment:
We could still return a 0-dimensional array here (pass `0,0` for the first 2 args), since that has 1 element.